src/sentry/static/sentry/app/views/organizationEventsV2/transactionView/minimap.tsx

```diff
@@ -0,0 +1,663 @@
+        {lastTick}
+        {this.renderCursorGuide(TIME_AXIS_HEIGHT)}
+        {this.renderDurationGuide()}
+      </TimeAxis>
+    );
+  };
+
+  render() {
+    return (
+      <MinimapContainer id="minimap-container">
+        <ActualMinimap trace={this.props.trace} />
+        <div
+          ref={this.props.minimapInteractiveRef}
+          style={{
+            width: '100%',
+            height: `${MINIMAP_HEIGHT + TIME_AXIS_HEIGHT}px`,
+            position: 'absolute',
+            left: 0,
+            top: 0,
+          }}
+          onMouseEnter={event => {
+            this.setState({
+              showCursorGuide: true,
+              mousePageX: event.pageX,
+            });
+          }}
+          onMouseLeave={() => {
+            this.setState({showCursorGuide: false, mousePageX: void 0});
+          }}
+          onMouseMove={event => {
+            this.setState({
+              showCursorGuide: true,
+              mousePageX: event.pageX,
+            });
+          }}
+        >
+          <InteractiveLayer>
+            {this.renderFog(this.props.dragProps)}
+            {this.renderCursorGuide(MINIMAP_HEIGHT)}
+            {this.renderViewHandles(this.props.dragProps)}
+          </InteractiveLayer>
+          {this.renderTimeAxis()}
+        </div>
+      </MinimapContainer>
+    );
+  }
+}
+
+class ActualMinimap extends React.PureComponent<{trace: ParsedTraceType}> {
+  drawMinimap = () => {
+    return this.renderRootSpan();
+  };
+
+  renderRootSpan = () => {
+    const {trace} = this.props;
+
+    const pickSpanBarColour = generateSpanColourPicker();
+
+    const generateBounds = boundsGenerator({
+      traceStartTimestamp: trace.traceStartTimestamp,
+      traceEndTimestamp: trace.traceEndTimestamp,
+      viewStart: 0,
+      viewEnd: 1,
+    });
+
+    const rootSpan: SpanType = {
+      trace_id: trace.traceID,
+      span_id: trace.rootSpanID,
+      start_timestamp: trace.traceStartTimestamp,
+      timestamp: trace.traceEndTimestamp,
+      data: {},
+    };
+
+    return this.renderSpan({
+      pickSpanBarColour,
+      generateBounds,
+      span: rootSpan,
+      spanID: trace.rootSpanID,
+      lookup: trace.lookup,
+    });
+  };
+
+  getBounds = (
+    bounds: SpanGeneratedBoundsType
+  ): {
+    left: string;
+    width: string;
+  } => {
+    switch (bounds.type) {
+      case 'TRACE_TIMESTAMPS_EQUAL':
+      case 'INVALID_VIEW_WINDOW': {
+        return {
+          left: toPercent(0),
+          width: '0px',
+        };
+      }
+
+      case 'TIMESTAMPS_EQUAL': {
+        return {
+          left: toPercent(bounds.start),
+          width: `${bounds.width}px`,
+        };
+      }
+      case 'TIMESTAMPS_REVERSED':
+      case 'TIMESTAMPS_STABLE': {
+        return {
+          left: toPercent(bounds.start),
+          width: toPercent(bounds.end - bounds.start),
+        };
+      }
+      default: {
+        const _exhaustiveCheck: never = bounds;
+        return _exhaustiveCheck;
```

Comment:
Should this be an exception?

Reply:
@markstory it doesn't need to be. 

I didn't add one because this file is consumed only by other typescript files. So we defer to the TS engine to ensure we've added all necessary the cases for the switch statement. This relies on `--strictNullChecks` being enabled.

AFAIK, there are two documented ways to do exhaustive checks for switch-case: https://www.typescriptlang.org/docs/handbook/advanced-types.html#exhaustiveness-checking